CharacterData.py

```diff
@@ -1,112 +1,190 @@
 class CharacterData(object):
+
 	"""
 	CharacterData contains constant values set by the game for a character
 
 	Attributes:
-		name: A string describing a character's name
-		game_class_options: A list of strings describing the in game classes
-			a character can choose from
-		growth_rate_class: A dicitonary mapping class names (from 
-			game_clas_options) onto GrowthRates objects  
-		base_stats: A dictionary mapping stat names to the character's base class stats
-		base_class: The class this character is in when they join in the game
+		__name : a str giving the character's name
+		__game_class_options : a list<str> describing all in game classes this
+			character can be
+		__growth_rate_class: a dict<str, GrowthRate> mapping class names onto 
+			GrowthRate objects associated with that character/class combination
+		__base_stats : a State describing the character's State when they are
+			first encountered
 	"""
 	
-	name = ""
-	game_class_options = []
-	growth_rate_class = {}
-	base_stats = {}
-	base_class = ""
-	base_level = 0
-	
 	class GrowthRates(object):
 		"""
-		GrowthRates wrap a dictioanry of individual growth rates for a given
+		GrowthRates wraps a dictionary of growth rates for a given 
 			character/class combination
 
 		Attributes:
-			rates: A dictionary mapping stats onto growth rates
+			__rates : a dict<Stat, float> mapping stats onto their growth rates
 		"""
-		rates = {}
 
+		def __init__(self, stat_values):
+			"""Inits GrowthRates with rates
+
+			Args:
+				stat_values : a list<float> containing the values associated
+					with Stats in the following order
+					[HP, Str, Mag, Skl, Spd, Lck, Def, Res]
+
+			Returns:
+				GrowthRates object
+			"""
+			pass
+
+		@property
+		def rates(self):
+			return self.__rates
+
+		if False:
+			"""
+			GrowthRates wrap a dictioanry of individual growth rates for a given
+				character/class combination
+
+			Attributes:
+				__rates: A dictionary mapping stats onto growth rates
+			"""
+			__rates = {}
+
+			def __init__(self, array):
+				"""Inits GrowthRates with rates"""
+				self.rates = {}
+				self.rates['HP'] = int(array[2])
+				self.rates['Str'] = int(array[3])
+				self.rates['Mag'] = int(array[4])
+				self.rates['Skl'] = int(array[5])
+				self.rates['Spd'] = int(array[6])
+				self.rates['Lck'] = int(array[7])
+				self.rates['Def'] = int(array[8])
+				self.rates['Res'] = int(array[9])
+				pass
```

Comment:
Logic will need to be re-used later with indices - 2 used instead of current array[] indices.